exchange/utils.go

```diff
@@ -302,14 +313,45 @@ func buildRequestExtForBidder(bidder string, requestExt json.RawMessage, request
 
 	// Update Ext With Prebid Json
 	extMap := make(map[string]json.RawMessage)
-	if err := json.Unmarshal(requestExt, &extMap); err != nil {
-		return nil, err
+	if len(requestExt) != 0 {
+		if err := json.Unmarshal(requestExt, &extMap); err != nil {
+			return nil, err
+		}
 	}
 	extMap["prebid"] = prebidJson
 
 	return json.Marshal(extMap)
 }
 
+func buildRequestExtAlternateBidderCodes(bidder string, accABC config.AlternateBidderCodes, reqABC *openrtb_ext.ExtAlternateBidderCodes) *openrtb_ext.ExtAlternateBidderCodes {
+	if reqABC != nil {
+		alternateBidderCodes := &openrtb_ext.ExtAlternateBidderCodes{
+			Enabled: reqABC.Enabled,
+		}
+		if bidderCodes, ok := reqABC.Bidders[bidder]; ok {
+			alternateBidderCodes.Bidders = map[string]openrtb_ext.ExtAdapterAlternateBidderCodes{
+				bidder: bidderCodes,
+			}
+		}
+		return alternateBidderCodes
+	}
+
+	alternateBidderCodes := &openrtb_ext.ExtAlternateBidderCodes{
+		Enabled: accABC.Enabled,
+	}
+	if bidderCodes, ok := accABC.Bidders[bidder]; ok {
+		alternateBidderCodes.Bidders = map[string]openrtb_ext.ExtAdapterAlternateBidderCodes{
+			bidder: bidderCodes,
+		}
+	}
+
+	if !alternateBidderCodes.Enabled && alternateBidderCodes.Bidders == nil {
```

Comment:
I'm wondering now if we should delete this if block. If we remove it, `ext.prebid.alternatebiddercodes` in the bidder request will contain exactly what's pertinent to the bidder in the account config with the lone exception being that `enabled` flags will be set to the default value false if not defined. This is the same behavior as if `ext.prebid.alternatebiddercodes` were to be populated on the bidder request with information pulled from the original request, assuming we remove the couple of `omitempty` tags I pointed out in my other comments.

Reply:
agreed. removed this code.